src/store/selectors/deck-filters.ts

```diff
@@ -0,0 +1,277 @@
+import { createSelector } from "reselect";
+
+import { assert } from "@/utils/assert";
+import { FACTION_ORDER } from "@/utils/constants";
+import { and, or } from "@/utils/fp";
+import uFuzzy from "@leeoniya/ufuzzy";
+import type { ResolvedDeck } from "../lib/types";
+import type { StoreState } from "../slices";
+import type { MultiselectFilter } from "../slices/lists.types";
+import { selectLocalDecks } from "./decks";
+
+import { capitalize } from "@/utils/formatting";
+import type {
+  DeckFiltersKey,
+  DeckPropertyName,
+  DeckValidity,
+  RangeMinMax,
+} from "../slices/deck-collection-filters.types";
+
+// Arbitrarily chosen for now
+const MATCHING_MAX_TOKEN_DISTANCE_DECKS = 4;
+
+export const selectDeckFilters = (state: StoreState) =>
+  state.deckFilters.filters;
+
+export const selectDeckFilterValue = createSelector(
+  selectDeckFilters,
+  (_, filter: DeckFiltersKey) => filter,
+  (filters, filter) => {
+    return filters[filter];
+  },
+);
+
+// Search
+export const selectDeckSearchTerm = (state: StoreState) =>
+  state.deckFilters.filters.search;
+
+export const selectSearchableTextInDecks = createSelector(
+  selectLocalDecks,
+  (decks) => {
+    // deck.investigator_name doesn't work for imported decks?
+    return decks.map(
+      (deck) =>
+        `${deck.name}|${deck.investigator_name || deck.cards.investigator.card.real_name}`,
```

Comment:
I think we should not use investigator_name here as only the real_name accessor can be i18n in the future